apollo-client/src/main/java/io/logz/apollo/clients/ApolloClient.java

```diff
@@ -236,4 +240,98 @@ public Notification getNotification(int id) throws ApolloClientException {
     public List<EnvironmentServiceGroupMap> getUndeployedServicesByAvailability(String availability, TimeUnit timeUnit, int duration) throws ApolloClientException, Exception {
         return genericApolloClient.getResult("/status/get-undeployed-services/avaiability/" + availability + "/time-unit/" + timeUnit + "/duration/" + duration, new TypeReference<List<EnvironmentServiceGroupMap>>() {});
     }
+
+    public EnvironmentsStack getEnvironmentsStack(int id) throws ApolloClientException {
+        return genericApolloClient.getResult("/environments-stack/" + id, new TypeReference<EnvironmentsStack>() {});
+    }
+
+    public List<EnvironmentsStack> getAllEnvironmentsStacks() throws ApolloClientException {
+        return genericApolloClient.getResult("/environments-stack", new TypeReference<List<EnvironmentsStack>>() {});
+    }
+
+    public EnvironmentsStack addEnvironmentToStack(int environmentId, int stackId) throws Exception {
+        String requestBody = Common.generateJson(
+                "environmentId", String.valueOf(environmentId),
+                "stackId", String.valueOf(stackId));
+        EnvironmentsStack environmentsStack = genericApolloClient.postAndGetResult("/environments-stack/environment", requestBody, new TypeReference<EnvironmentsStack>() {});
+        return environmentsStack;
+    }
+
+    public EnvironmentsStack addEnvironmentsStack(EnvironmentsStack environmentsStack) throws Exception {
+        String environmentsCsv = StringUtils.join(environmentsStack.getEnvironments(), ",");
```

Comment:
Lets create a function for that please?

Reply:
I already created one in apollo-backend/StringParser, but it creates circular dependency between apollo-backend and apollo-client

Reply:
Hence the create a function comment and not include apollo-backend